aten/src/ATen/cpu/vec256/vec256_qint.h

```diff
@@ -413,12 +413,12 @@ Vec256<c10::qint32> inline operator*(
   return _mm256_mullo_epi32(a, b);
 #else
   // Pray the compiler can autovectorize this
-  int32_t a_vals[a.size()];
-  int32_t b_vals[b.size()];
+  int32_t a_vals[std::decay_t<decltype(a)>::size()];
```

Comment:
I wonder if it's better to use `std::array` here, to avoid any `constexpr` interpretation ambiguity.

Reply:
Great catch! Imho `std:array` here would work just fine

Reply:
I second the `std::array` proposal here for sure.